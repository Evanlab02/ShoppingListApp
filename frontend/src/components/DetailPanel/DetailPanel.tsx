import React from "react";
import { DetailPanelProps } from "./helpers/interfaces";
import "./styles/DetailPanel.scss"

/**
 * Detail panel component.
 * 
 * @returns Detail panel component
 */
export default function DetailPanel(props: DetailPanelProps) {

    const { records } = props;

    const renderedRecentItems = records.map((record, index) => {
        return (
            <React.Fragment key={`${record.name}-${record.price}-${index}`}>
                <hr />
                <div className="recent-item">
                    <h4>{record.name}</h4>
                    <h4>R{record.price}</h4>
                </div>
            </React.Fragment>
        );
    });

    return (
        <div className="detail-panel" data-testid="sa-detail-panel">
            <h4>Recent Items</h4>
            {renderedRecentItems}
        </div>
    );
}