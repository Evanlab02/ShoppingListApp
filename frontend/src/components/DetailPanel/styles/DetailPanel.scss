.detail-panel {
    background-color: #ffffff;
    border-radius: 0.5rem;
    padding: 1rem;
    margin: 0.2rem;

    h4 {
        margin: 0rem;
        padding: 0rem;
        padding-bottom: 0.5rem;
    }

    .recent-item {
        display: flex;
        justify-content: space-between;
        align-items: center;
        height: 60px;
    }
}