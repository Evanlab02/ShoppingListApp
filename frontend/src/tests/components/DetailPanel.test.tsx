import { beforeEach, expect, test } from "vitest";
import { render } from "@testing-library/react";
import DetailPanel from "../../components/DetailPanel/DetailPanel";

beforeEach(() => {
	document.getElementsByTagName("html")[0].innerHTML = "";
});

test("Empty detail panel renders correctly.", async () => {
    const { findByTestId } = render(
        <DetailPanel
            records={[]}
        />
	);

	const panel = await findByTestId("sa-detail-panel");
	expect(panel).toMatchSnapshot();
});

test("Populated detail panel renders correctly.", async () => {
    const { findByTestId } = render(
        <DetailPanel
            records={[
                {name: "1", price: 100},
                {name: "2", price: 200},
                {name: "3", price: 300},
                {name: "4", price: 400},
                {name: "5", price: 500}
            ]}
        />
	);

	const panel = await findByTestId("sa-detail-panel");
	expect(panel).toMatchSnapshot();
});