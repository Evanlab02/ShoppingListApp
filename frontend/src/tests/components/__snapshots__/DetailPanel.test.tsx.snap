// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Empty detail panel renders correctly. 1`] = `
<div
  class="detail-panel"
  data-testid="sa-detail-panel"
>
  <h4>
    Recent Items
  </h4>
</div>
`;

exports[`Populated detail panel renders correctly. 1`] = `
<div
  class="detail-panel"
  data-testid="sa-detail-panel"
>
  <h4>
    Recent Items
  </h4>
  <hr />
  <div
    class="recent-item"
  >
    <h4>
      1
    </h4>
    <h4>
      R
      100
    </h4>
  </div>
  <hr />
  <div
    class="recent-item"
  >
    <h4>
      2
    </h4>
    <h4>
      R
      200
    </h4>
  </div>
  <hr />
  <div
    class="recent-item"
  >
    <h4>
      3
    </h4>
    <h4>
      R
      300
    </h4>
  </div>
  <hr />
  <div
    class="recent-item"
  >
    <h4>
      4
    </h4>
    <h4>
      R
      400
    </h4>
  </div>
  <hr />
  <div
    class="recent-item"
  >
    <h4>
      5
    </h4>
    <h4>
      R
      500
    </h4>
  </div>
</div>
`;
